app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp

```diff
@@ -1061,6 +1072,46 @@ struct DeviceDelegateOculusVR::State {
 
         const bool thumbRest = (controllerState.inputState.Touches & ovrTouch_ThumbUp) != 0;
         controller->SetButtonState(controllerState.index, ControllerDelegate::BUTTON_OTHERS, 5, thumbRest, thumbRest);
+
+        // Only Oculus 6 DOF (Quest) has haptic feedback.
+        controller->SetHapticCount(controllerState.index, 1);
```

Comment:
`UpdateControllers` is becoming very big. Can you move the haptics code to a separate function? (e.g. `UpdateHaptics`)